labman/db/tests/integration_tests.py

```diff
@@ -0,0 +1,214 @@
+
+        # Make sure that the user didn't pass more samples than wells
+        if i > num_rows:
+            break
+
+        sp_process.update_well(i, j, sample)
+
+    sample_plate = sp_process.plate
+    return sp_process, sample_plate
+
+
+def create_gdna_extraction_process(user, plate):
+    kingfisher = Equipment(11)
+    epmotion = Equipment(6)
+    epmotion_tool = Equipment(15)
+    extraction_kit = ReagentComposition(1)
+    ext_process = GDNAExtractionProcess.create(
+        user, plate, kingfisher, epmotion, epmotion_tool, extraction_kit, 100,
+        'GDNA test plate %s' % datetime.now())
+    gdna_plate = ext_process.plates[0]
+    return ext_process, gdna_plate
+
+
+def create_amplicon_prep(user, plate):
+    primer_plate = Plate(11)
+    epmotion = Equipment(6)
+    master_mix = ReagentComposition(2)
+    water_lot = ReagentComposition(3)
+    epmotion_tool_tm300 = Equipment(16)
+    epmotion_tool_tm50 = Equipment(17)
+    amplicon_process = LibraryPrep16SProcess.create(
+        user, plate, primer_plate, 'Amplicon test plate %s' % datetime.now(),
+        epmotion, epmotion_tool_tm300, epmotion_tool_tm50, master_mix,
+        water_lot, 75,)
+    amplicon_plate = amplicon_process.plates[0]
+    return amplicon_process, amplicon_plate
+
+
+def create_compression_process(user, gdna_plates):
+    comp_process = GDNAPlateCompressionProcess.create(
+        user, gdna_plates, 'Compressed test plate %s' % datetime.now(),
+        Equipment(6))
+    compressed_plate = comp_process.plates[0]
+    return comp_process, compressed_plate
+
+
+def create_quantification_process(user, plate):
+    plate_config = plate.plate_configuration
+    concentrations = np.around(
+        np.random.rand(plate_config.num_rows, plate_config.num_columns), 6)
+    quant_process = QuantificationProcess.create(user, plate, concentrations)
+    return quant_process
+
+
+def create_pool_quantification_process(user, pools):
+    concentrations = np.around(np.random.rand(len(pools)), 6)
+    concentrations = [{'composition': p, 'concentration': c}
+                      for p, c in zip(pools, concentrations)]
+    return QuantificationProcess.create_manual(user, concentrations)
+
+
+def create_normalization_process(user, quant_process):
+    water = ReagentComposition(3)
+    norm_process = NormalizationProcess.create(
+        user, quant_process, water,
+        'Normalized test plate %s' % datetime.now())
+    norm_plate = norm_process.plates[0]
+    return norm_process, norm_plate
+
+
+def create_shotgun_process(user, norm_plate):
+    kappa = ReagentComposition(4)
+    stub = ReagentComposition(5)
+    shotgun_process = LibraryPrepShotgunProcess.create(
+        user, norm_plate, 'Test Shotgun Library %s' % datetime.now(), kappa,
+        stub, 4000, Plate(19), Plate(20))
+    shotgun_plate = shotgun_process.plates[0]
+    return shotgun_process, shotgun_plate
+
+
+def create_plate_pool_process(user, shotgun_quant_process, shotgun_plate,
```

Comment:
I think this is generic for shotgun and amplicon pools, right? so maybe just `quant_process` and `plate`?

Reply:
Good point - I was copy/pasting code around :D